src/client/hud.cpp

```diff
@@ -377,15 +379,19 @@ void Hud::drawLuaElements(const v3s16 &camera_offset)
 				std::wstring text = unescape_translate(utf8_to_wide(e->text));
 				core::dimension2d<u32> textsize = textfont->getDimension(text.c_str());
 
-				v2s32 offset((e->align.X - 1.0) * (textsize.Width / 2),
-				             (e->align.Y - 1.0) * (textsize.Height / 2));
+				v2s32 offset(0, (e->align.Y - 1.0) * (textsize.Height / 2));
 				core::rect<s32> size(0, 0, e->scale.X * m_scale_factor,
-				                     text_height * e->scale.Y * m_scale_factor);
+									 text_height * e->scale.Y * m_scale_factor);
 				v2s32 offs(e->offset.X * m_scale_factor,
-				           e->offset.Y * m_scale_factor);
-
+						   e->offset.Y * m_scale_factor);
+				std::wstringstream wss(text);
+				std::wstring line;
+				while (std::getline(wss, line, L'\n'))
```

Comment:
Can't line splitting be avoided here if align is the default value (0?),

Reply:
> Can't line splitting be avoided here if align is the default value (0?), I think that's preferable if possible.

Line splitting can be avoided if align is -1 (left justified). Which poses yet another issue: People might rely on the broken behavior to left-justify multiline text HUDs without align values.

Reply:
The old and new left-aligned behavior is perfectly identical isn't it?
I don't quite get which case you're describing.

Reply:
No matter the value of align, due to the bug described by Wuzzy, all multiline text used to be left-justified.

This fix now centers text with align.x = 0. The default value for align is {x=0, y=0}, so if some modders omit align, they would now get centered text instead of the left justified text they might (wrongly) expect.

Reply:
If it's documented that align=0 creates centered text that's fine.

Reply:
Alright. I'd love to just split the text on init, but that would require some kind of proper inheritance / class hierarchy instead of the current thing. I wouldn't want to add a `std::vector<std::string> lines` field to all HUD elements.

I doubt that `align = 0` would be much of an improvement in terms of performance - it will just leave Irrlicht / Freetype? to do the line splitting.

Reply:
Okay then.